duktape/src/main/jni/DuktapeContext.cpp

```diff
@@ -219,3 +228,52 @@ void DuktapeContext::bindInstance(JNIEnv *env, jstring name, jobject object, job
   // Pop the Duktape global object off the stack.
   duk_pop(m_context);
 }
+
+jobject DuktapeContext::call(JNIEnv *env, jstring name, jobject method, jobjectArray args) {
+  duk_push_global_object(m_context);
+  const JString instanceName(env, name);
+  if (!duk_get_prop_string(m_context, -1, instanceName)) {
+    duk_pop(m_context);
+    const jclass illegalArgumentException = env->FindClass("java/lang/IllegalArgumentException");
+    const std::string message = "A global object called " + instanceName.str() + " was not found";
```

Comment:
I’ve got a pass-by-value / pass-by-reference question for you. Suppose I do the following strange sequence of calls:

1. JavaScript:
```
var x = {
  greet: function() { return "hello"; }
}
```

2. Java:
```
Greeter greeter = duktape.proxy("x", Greeter.class);
```

3. JavaScript
```
x = {
  greet: function() { return "howdy"; }
}
```

4. Java:
```
System.out.println(greeter.greet());
```

Does it print `hello` or `howdy` ? Effectively I’m asking whether we should bind `x` at the time of the call to `proxy()`, or if we should bind it upon each method call.

Doing it eagerly means we have a persistent reference across heaps that will need to be cleared, either manually, or by a side effect of the Java finalizer (which has concurrency problems). This is more difficult technically!

But doing it eagerly _also_ means that we can validate that `x` exists eagerly, and even if `x` is reassigned you get the value at the time of the `proxy()` call. There’s a tiny potential performance benefit because we will will look up `x` by something stronger than its name.